pkg/api/handlers/compat/containers_attach.go

```diff
@@ -92,30 +92,29 @@ func AttachContainer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	idleTracker := r.Context().Value("idletracker").(*idle.Tracker)
-	hijackChan := make(chan bool, 1)
+	logErr := func(e error) {
+		logrus.Error(errors.Wrapf(e, "error attaching to container %s", ctr.ID()))
+	}
 
 	// Perform HTTP attach.
 	// HTTPAttach will handle everything about the connection from here on
 	// (including closing it and writing errors to it).
-	if err := ctr.HTTPAttach(r, w, streams, detachKeys, nil, query.Stream, query.Logs, hijackChan); err != nil {
-		hijackComplete := <-hijackChan
+	hijackChan := make(chan bool, 1)
+	err = ctr.HTTPAttach(r, w, streams, detachKeys, nil, query.Stream, query.Logs, hijackChan)
 
-		// We can't really do anything about errors anymore. HTTPAttach
-		// should be writing them to the connection.
-		logrus.Errorf("Error attaching to container %s: %v", ctr.ID(), err)
+	if <-hijackChan {
```

Comment:
I don't think this is going to work. `HTTPAttach` will not return until attach is already complete. We need to have a go func before this to handle the channel, I think.

Reply:
@mheon It's working, at least for the case below and the gitlab-runner use case.   We don't want the IdleTracker to kick in until HTTPAttach() is finished.

```
$ podman-remote run... /bin/bash
deadbeef
$ podman-remote attach deadbeef
[root@deadbeef /]# ls
...
[root@deadbeef /]# exit                                                                                                         
                                               
```